### ISSUE TITLE
Tabs, TaNav, TaTitle and Tab - Ensure page tabs provide state and role - (2036570288)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5607]
### Module: 
 32 Tabs, Tab Nav, Tab Title and Tab 

### Violation: 
 Ensure page tabs provide state and role 

### WCAG Reference: 
- [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570288 

### Severity: 
 7 

### Description: 
 [Issue]
There are tab panels without appropriate role association information. Examples include:

- Rain
- Snowy rain
- Thunderstorms

[User Impact]
Screen reader users will be unable to determine that these controls reveal panels of content and which panel is currently revealed.

[Code Reference] 
 ```
<calcite-tab-title active="" icon-start="rain" aria-selected="true" id="calcite-tab-title-bfa8f1a2-1f01-98eb-475a-9e530dee965b" role="tab" tabindex="0" selected="" layout="inline" position="top" scale="m" calcite-hydrated="" aria-controls="calcite-tab-title-7fd48e2a-096f-de25-7475-e9ca53847ac4"></calcite-tab-title>
(...)
<calcite-tab active="" id="calcite-tab-title-7fd48e2a-096f-de25-7475-e9ca53847ac4" scale="m" calcite-hydrated="" aria-labelledby="calcite-tab-title-bfa8f1a2-1f01-98eb-475a-9e530dee965b">Rain Content</calcite-tab>
 ``` 

### Note: 
 [Suggestion]
Ensure page tabs provide state and role.

For tabs, the following information is expected:
- Each panel container must have role="tabpanel".
- Each tab must have aria-controls set to the ID of its corresponding tabpanel.

[Compliant Code Example] 
 ```
<calcite-tab-title aria-label="rain" aria-control="calcite-tab-title-7fd48e2a-096f-de25-7475-e9ca53847ac4" active="" icon-start="rain" aria-selected="true" id="calcite-tab-title-bfa8f1a2-1f01-98eb-475a-9e530dee965b" role="tab" tabindex="0" selected="" layout="inline" position="top" scale="m" calcite-hydrated="" aria-controls="calcite-tab-title-7fd48e2a-096f-de25-7475-e9ca53847ac4"></calcite-tab-title>
(...)
<calcite-tab active="" role="tabpanel" id="calcite-tab-title-7fd48e2a-096f-de25-7475-e9ca53847ac4" scale="m" calcite-hydrated="" aria-labelledby="calcite-tab-title-bfa8f1a2-1f01-98eb-475a-9e530dee965b">Rain Content</calcite-tab>
 ```
 

### Media Type: 
 Page Tabs 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.
